languageserver/integration/commands.go

```diff
@@ -219,41 +342,41 @@ func (i *FlowIntegration) executeScript(conn protocol.Conn, args ...interface{})
 // There should be exactly 1 argument:
 //   * the address of the new active account
 func (i *FlowIntegration) switchActiveAccount(_ protocol.Conn, args ...interface{}) (interface{}, error) {
-
-	err := server.CheckCommandArgumentCount(args, 1)
+	err := server.CheckCommandArgumentCount(args, 2)
 	if err != nil {
 		return nil, err
 	}
 
-	addrHex, ok := args[0].(string)
+	name, ok := args[0].(string)
 	if !ok {
-		return nil, errors.New("invalid argument")
+		return nil, errors.New("invalid name argument")
 	}
-	addr := flow.HexToAddress(addrHex)
 
-	_, ok = i.accounts[addr]
+	addressHex, ok := args[1].(string)
 	if !ok {
-		return nil, errors.New("cannot set active account that does not exist")
+		return nil, errors.New("invalid address argument")
 	}
+	address := flow.HexToAddress(addressHex)
 
-	i.activeAddress = addr
+	i.activeAccount = ClientAccount{
+		Name:    name,
+		Address: address,
+	}
 	return nil, nil
 }
 
 // createAccount creates a new account and returns its address.
 func (i *FlowIntegration) createAccount(conn protocol.Conn, args ...interface{}) (interface{}, error) {
 
-	err := server.CheckCommandArgumentCount(args, 0)
-	if err != nil {
-		return nil, err
-	}
+	address, err := i.createAccountHelper(conn)
+	clientAccount, err := i.storeAccountHelper(conn, address)
 
-	addr, err := i.createAccountHelper(conn)
-	if err != nil {
-		return nil, err
-	}
+	conn.ShowMessage(&protocol.ShowMessageParams{
+		Type:    protocol.Info,
+		Message: fmt.Sprintf("New account %s(0x%s) created", clientAccount.Name, address.String()),
+	})
 
-	return addr, nil
+	return clientAccount, err
```

Comment:
```suggestion
	return clientAccount, nil
```

Better to check the errors above and return early if there's an error. Then you can just return `nil` here